src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/CompletionListCache.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Razor.Completion;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Completion
+{
+    internal sealed class CompletionListCache
```

Comment:
@allisonchou I started looking at the the [Roslyn equivalent of this](https://github.com/dotnet/roslyn/blob/master/src/Features/LanguageServer/Protocol/Handler/Completion/CompletionListCache.cs) that you did and saw that it was async. I choose to take a different route and leave it synchronous but wanted to ask. Is there a reason why the Roslyn version is async?

Reply:
Hm, that's a good catch! I think it could be synchronous actually, and that would probably simplify the code as well. I've filed an issue on the Roslyn side here - https://github.com/dotnet/roslyn/issues/50418